apps/src/schoolInfo/SchoolInfoInterstitial.jsx

```diff
@@ -1,378 +1,233 @@
-    });
-    if (!isValid) {
-      return;
+    // if schoolId exists, don't disable unless selected school is not in the schools list
+    if (
+      schoolInfo.schoolId &&
+      schoolInfo.schoolsList.some(({value}) => schoolInfo.schoolId === value)
+    ) {
+      return false;
     }
+    // disable by default
+    return true;
+  }, [
+    schoolInfo.country,
+    schoolInfo.schoolId,
+    schoolInfo.schoolZip,
+    schoolInfo.schoolName,
+    schoolInfo.schoolsList,
+  ]);
+
+  const handleSchoolInfoSubmit = async () => {
+    const hasNcesId =
+      schoolInfo.schoolId &&
+      !NON_SCHOOL_OPTIONS_ARRAY.includes(schoolInfo.schoolId);
     analyticsReporter.sendEvent(
       EVENTS.SCHOOL_INTERSTITIAL_SUBMIT,
       {
-        hasNcesId:
-          this.state.ncesSchoolId && this.state.ncesSchoolId !== '-1'
-            ? 'true'
-            : 'false',
-        attempt: this.state.showSchoolInfoUnknownError ? 2 : 1,
+        hasNcesId: hasNcesId.toString(),
+        attempt: showSchoolInfoUnknownError ? 2 : 1,
       },
       PLATFORMS.BOTH
     );
 
-    const schoolData = this.buildSchoolData();
-    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
-    $.post({
-      url: formUrl + '.json',
-      dataType: 'json',
-      data: {
-        _method: 'patch',
-        [authTokenName]: authTokenValue,
-        ...schoolData,
-      },
-    })
-      .done(() => {
-        analyticsReporter.sendEvent(
-          EVENTS.SCHOOL_INTERSTITIAL_SAVE_SUCCESS,
-          {
-            attempt: this.state.showSchoolInfoUnknownError ? 2 : 1,
-          },
-          PLATFORMS.BOTH
-        );
+    try {
+      await updateSchoolInfo({
+        formUrl,
+        authTokenName,
+        authTokenValue,
+        schoolId: schoolInfo.schoolId,
+        country: schoolInfo.country,
+        schoolName: schoolInfo.schoolName,
+        schoolZip: schoolInfo.schoolZip,
+      });
 
-        this.props.onClose();
-      })
-      .fail(() => {
-        analyticsReporter.sendEvent(
-          EVENTS.SCHOOL_INTERSTITIAL_SAVE_FAILURE,
-          {
-            attempt: this.state.showSchoolInfoUnknownError ? 2 : 1,
-          },
-          PLATFORMS.BOTH
-        );
+      analyticsReporter.sendEvent(
+        EVENTS.SCHOOL_INTERSTITIAL_SAVE_SUCCESS,
+        {
+          attempt: showSchoolInfoUnknownError ? 2 : 1,
+        },
+        PLATFORMS.BOTH
+      );
 
-        if (!this.state.showSchoolInfoUnknownError) {
-          // First failure, display error message and give the teacher a chance
-          // to try again.
-          this.setState({showSchoolInfoUnknownError: true});
-        } else {
-          // We already failed once, let's not block the teacher any longer.
-          this.props.onClose();
-        }
-      });
+      onClose();
+    } catch (error) {
+      analyticsReporter.sendEvent(
+        EVENTS.SCHOOL_INTERSTITIAL_SAVE_FAILURE,
+        {
+          attempt: showSchoolInfoUnknownError ? 2 : 1,
```

Comment:
Should we count attempts rather than assuming?

Reply:
haha, this is one of the very few things I actually didn't change from the original implementation 😅

Reply:
LOL sorry, then definitely no need to fix 😅